### ISSUE TITLE
Assorted small issues with the Mangalist.

### ISSUE THREAD

**Issue by Weefsat** [doutatsu/kenmei#14]
- [ ] Manga that have upcoming releases show them as latest chapter. They should show the last accessible chapter instead (e.g. Satanophany releases chapter 55 in 21 hours, should show chapter 54 instead [https://mangadex.org/manga/21119](url) ).
![Satanophany](https://user-images.githubusercontent.com/39443093/63214671-65efc180-c11b-11e9-8604-45659cf72d9c.png)

- [x] Some manga show and link incorrect latest chapters. (e.g. Fear shows chapter 1, Dead Tube shows 0, even though they are released way past that). The released date works properly for those titles.
[https://mangadex.org/manga/13525](url), [https://mangadex.org/manga/14947](url). 
![fear](https://user-images.githubusercontent.com/39443093/63214730-5de45180-c11c-11e9-807b-17bbc97a3195.png)

- [x] Some manga have incorrect names (e.g. "Pokémon - Dating a Team Magma Grunt (Doujinshi)" shows up as "Pok&eacute;mon - Dating a Team Magma Grunt (Doujinshi)"). [https://mangadex.org/manga/14246](url)
![pokemon](https://user-images.githubusercontent.com/39443093/63214703-f201e900-c11b-11e9-915f-1bd77b273598.png)

- [ ] Oneshots don't show any chapter number, should say something like oneshot instead. (e.g. Yoroshiku! Junjou Daishou). [https://mangadex.org/manga/35850](url)
![yoroshikou](https://user-images.githubusercontent.com/39443093/63214719-2bd2ef80-c11c-11e9-9eba-6c29b941672f.png)

- [x] Manga without entries show no chapter number or release date, should show something like no chapters released instead (e.g. Wangtangchang the Yurutto Hunter) [https://mangadex.org/manga/24214](url)
![Wangtangchang](https://user-images.githubusercontent.com/39443093/63214694-bd8e2d00-c11b-11e9-9d74-5befe2457218.png)

I hope some of this helps.

**Comment by doutatsu** [comment]
Thanks a lot for the detailed testing @Weefsat, this is super helpful, thank you for taking the time.

> Manga without entries show no chapter number or release date

That was intentional, quick way to work around those and I will definitely improve how it is shown. 

> Manga that has upcoming releases show them as the latest chapter

This is an interesting one. I personally find that I want to know about upcoming releases, so I can go and be redirected to the group release, instead of waiting. 

I am thinking of either having an option in settings, whatever you want to see the actual latest chapter or the latest available to read. Or showing the actual latest, but with an info popup detailing its released in timer. I am gravitating more for leaving it up to the end-user to choose which one they prefer.

---

The rest I wasn't aware of and will fix in the near future!

**Comment by Weefsat** [comment]
Glad I could help, and thanks working so hard on this!

>  I personally find that I want to know about upcoming releases, so I can go and be redirected to the group release, instead of waiting.

In general I do too, but often groups block the chapters behind a paywall or use the upcoming tab as a release schedule announcement without having the chapters available on their site. I hate having chapters marked as unread and having no way to actually read them. I understand that is a low priority issue. It would be fixed by an option in the settings.

I have 2 more issues to add.

1. Some manga don't show chapters (e.g. Kichikujima [https://mangadex.org/title/20176/kichikujima](url)).
![Kichikujima](https://user-images.githubusercontent.com/39443093/63228653-3c579880-c1f6-11e9-807c-9a305851b922.png)

2. When sorting by name, names that start with lowercase letters are sorted to the bottom of the alphabet. (e.g. nacun [https://mangadex.org/title/15864/nacun](url))
![sorting](https://user-images.githubusercontent.com/39443093/63228686-b4be5980-c1f6-11e9-94f9-980dba7ef172.png)

**Comment by doutatsu** [comment]
> Some manga have incorrect names

This issue has been resolved in #27. I've actually noticed that Trackr.moe had the same issue, so glad to know we've already got some improvements over it 😄 

If you don't mind, I'll update your messages to be a checklist instead, so its easier to track what has been addressed

**Comment by SaltatorMortis** [comment]
>      Manga that have upcoming releases show them as latest chapter....

maybe add in addition to a global/default option add a way to change/overwrite this option on a novel by novel basis

**Comment by doutatsu** [comment]
> maybe add in addition to a global/default option add a way to change/overwrite this option on a novel by novel basis

What are the series you'd overwrite the default setting? Would be good to know an example. I can see it making sense to disable all or enable all, not keeping them for some?

**Comment by SaltatorMortis** [comment]
huh after a closer look the novel that i could name isnt on mangadex anymore,
and this was my last novel with prerelease...
sorry i havnt read mangas activly the last few months..
well im the type of guy that trys to run the web without adblocker,
so basicly i would use this on websites that run maladverts/have a bad ad practice

tl;dr
generally this would be a nice to have feature but no need to have i would be happy with any implementation

**Comment by doutatsu** [comment]
@SaltatorMortis I see, so you are saying you wouldn't want to visit some of the websites, to check when next chapter is released, because of the ads. So knowing when they are coming out, would be still useful. But you are still getting a notification for when it is released, so I wonder if its still needed?

I can see it being done not on series by series basis, but provider by provider. So you could disable it on MangaDex, but say enable on MangaKakalot. Although I am not yet sure how many places you'll even have that functionality. From what I've seen, only MangaDex has such functionality to beign with, while others just released the chapter, with maybe having schedule specified somewhere on the website

**Comment by SaltatorMortis** [comment]
jeah implement simply a global/provider
with a littlebit self discipline a more granular control is not nessesary

**Comment by Weefsat** [comment]
Manga list shows the latest uploaded chapter, as the latest, instead of showing the latest story-wise. (e.g. Toujuushi Bestialious rereleased chapter 2 in higher quality. The list now shows chapter 2 as latest, even though the manga is released all the way to chapter 5.) https://mangadex.org/manga/1271
![Toujuushi](https://user-images.githubusercontent.com/39443093/66308439-78aa9980-e907-11e9-8eed-adffa072f949.png)

**Comment by doutatsu** [comment]
> Manga list shows the latest uploaded chapter, as the latest, instead of showing the latest story-wise

Yup, this one I am aware off and just went with upload time for now, but I will, of course, be implementing a better system that shows _actual_ next chapter.

**Comment by doutatsu** [comment]
> Some manga show and link incorrect latest chapters
This has now been addressed. There are a lot of edge cases, and I have an involved algorithm to find the latest chapter, so if you still find some that are incorrect, let me know. 

P.S. As of writing this, I am still running an update on _all_ the series currently in the DB (3000+) which takes about an hour, so expect to see things fixed in your own list after that time passes

**Comment by Weefsat** [comment]
For me, there are now many manga that incorrectly display the latest chapter. I believe there are more than before? Some examples below.
![Untitled](https://user-images.githubusercontent.com/39443093/68081622-570ed600-fe11-11e9-86b0-9c412ea5b8e3.png)
(https://mangadex.org/manga/11037, https://mangadex.org/manga/3491, https://mangadex.org/manga/9641, https://mangadex.org/manga/2256, https://mangadex.org/manga/19588, https://mangadex.org/manga/12306)

I can give you a more complete list if needed.

**Comment by doutatsu** [comment]
@Weefsat Haha yeah, I pulled the trigger a bit too early. There was a small bug for specific types of series, which I've fixed. This has all been fixed now and I checked all these links and they all good now. Let me know if you still some that have wrong latest chapter

**Comment by doutatsu** [comment]
I am closing this issue in favour of #77
